src/lib.rs

```diff
@@ -298,24 +298,54 @@ fn expand_to_file(
             .arg(&dest)
             .current_dir(cwd);
 
-        let res = process.status();
-        if allow_failure {
-            if let Err(err) = res {
+        handle_rustfmt_result(process.status(), &dest, allow_failure)?;
```

Comment:
I'd move the actual rust process spawning and waiting for exit code into the function as well

Reply:
Ok, is this what you have in mind?

``` rust
     if let RustFmt::Yes {
        channel,
        edition,
        allow_failure,
    } = rustfmt
    {
        format_rust_file(channel, edition, &dest, cwd, allow_failure)?;
    }

    let dest = dest.display().to_string();
    Ok(quote! {
        include!( #dest );
    })
}

fn format_rust_file(...) {
...
}

```